packages/block-library/src/navigation-link/edit.js

```diff
@@ -114,6 +113,41 @@ function NavigationLinkEdit( {
 		selection.addRange( range );
 	}
 
+	async function handleCreatePage( pageTitle ) {
+		const type = 'page';
+		const page = await saveEntityRecord( 'postType', type, {
+			title: pageTitle,
+			status: 'publish',
```

Comment:
Just to confirm; we're certain that it makes sense to publish this page immediately? And not save as draft? I guess normally my concern there would be if the site is set up to include new pages automatically in navigation menus, but I suppose that may be intentional or at least covered as part of this block?

Reply:
@obenland What are your thoughts on this?

Reply:
I think this needs some design input. It depends on what the expected UX for this is. However, my 2 cents is that in either case (draft or publish statuses) we'll have a weird situation:

1) the menu is live, the new page is draft, results in a broken link
2) the menu is live, the new page is live, results in an empty page link

(2) sounds like a more preferable state but I personally think that publishing a page while working on some menu is risky and unexpected.

IMO, the best way would be similar to the missing feature of "add new pages automatically", that means a flow like:

- add a new menu item with a non-existent page
- create a page with draft status and title from the menu item
- save whatever the menu is in (a post, a template part etc.)
- when rendering the menu check the status of its page links and do not render anything other than those with "publish" status

Reply:
> create a page with draft status and title from the menu item

> when rendering the menu check the status of its page links and do not render anything other than those with "publish" status

It will bring an inconsistency between the editor-canvas and the front-end, not only for the new page created on the fly but with all not published pages.
If I'm not wrong, it possible to add only published pages in the navigation menu, is it correct?

Reply:
I feel like this is definitely something worth revisiting after this PR was merged

Reply:
I'll raise an issue.